src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -157,10 +157,19 @@ var ReactDOM = {
     }
     // Unsound duck typing.
     const component = (componentOrElement : any);
-    if (component.nodeType === 1) {
+    if (component.nodeType === 1) { // Element
       return component;
     }
-    return DOMRenderer.findHostInstance(component);
+    if (component.updater) { // Component
+      const whileRendering = component._reactInternalInstance.progressedChild;
+      if (!component.updater.isMounted(component) && whileRendering) {
+        throw new Error('findDOMNode was called on an unmounted component.');
+      }
+      return DOMRenderer.findHostInstance(component);
+    }
+    throw new Error(
```

Comment:
This is a different duck typing than we currently have. We check for `render` to determine this error.